readthedocs/search/faceted_search.py

```diff
@@ -1,44 +1,91 @@
+import logging
+
 from elasticsearch_dsl import FacetedSearch, TermsFacet
+from elasticsearch_dsl.query import Bool, SimpleQueryString
+
+from readthedocs.search.documents import PageDocument, ProjectDocument
+from readthedocs.search.signals import before_file_search, before_project_search
+
+log = logging.getLogger(__name__)
 
 
 class RTDFacetedSearch(FacetedSearch):
 
-    """Overwrite the initialization in order too meet our needs"""
+    """Overwrite the initialization in order too meet our needs."""
 
     # TODO: Remove the overwrite when the elastic/elasticsearch-dsl-py#916
     # See more: https://github.com/elastic/elasticsearch-dsl-py/issues/916
 
-    def __init__(self, using, index, doc_types, model, fields=None, **kwargs):
-        self.using = using
-        self.index = index
-        self.doc_types = doc_types
-        self._model = model
-        if fields:
-            self.fields = fields
+    def __init__(self, user, **kwargs):
+        self.user = user
         super(RTDFacetedSearch, self).__init__(**kwargs)
 
+    def search(self):
+        """
+        Filter out full content on search results.
+
+        This was causing all of the indexed content to be returned, which was
+        never used on the client side.
+        """
+        s = super().search()
+        s = s.source(exclude=['content', 'headers'])
+        resp = self.signal.send(sender=self, user=self.user, search=s)
+        if resp:
+            # Signal return a search object
```

Comment:
There are some docs about this: https://docs.djangoproject.com/en/1.11/topics/signals/#django.dispatch.Signal.send

We should be careful about how we manipulate the result considering that it's a list.

Reply:
Yea, I tried to manipulate the object that was passed, but it didn't work in my testing. I'm not sure another good way to implement this.

Reply:
I think the line `resp[0][1]` will break if there are more signals connected to `before_file_search` signal, since the result will depend on which one is executed first.

Why are we using a signal for this? I suppose that it's because we need to override this on .com. In that case, isn't it better to make it a method and override the class using our SettingsOverride system?

> I found that this signal is connected only to one receiver on each project.

Reply:
Yea, having a class override on the .com is probably a cleaner solution here. I will work on this.

Reply:
Pushed up a new version with this approach.